3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js

```diff
@@ -15,12 +15,18 @@ const rollDice = require('../../helpers/pokerDiceRoller');
 function rollTheDices() {
   // TODO Refactor this function
   const dices = [1, 2, 3, 4, 5];
-  return rollDice(1);
+  const combinedPromise = dices.map((element) => rollDice(element));
+
+  Promise.all(combinedPromise)
+    .then((results) => {
+      console.log('Resolved!', results);
+    })
+    .catch((error) => console.log('Rejected!', error.message));
 }
 
-rollTheDices()
-  .then((results) => console.log('Resolved!', results))
-  .catch((error) => console.log('Rejected!', error.message));
+rollTheDices();
+// .then((results) => console.log('Resolved!', results))
+// .catch((error) => console.log('Rejected!', error.message));
```

Comment:
You should use the code from lines 27-29 as they were provided. That means that `rollDices()` should return a promise, i.e. the promise returned by `Promise.all()`:

```js
function rollTheDice() {
  ...
  return Promise.all(combinedPromise);
}
```

Reply:
Implemented.